services/frontend-service/src/ui/ActionDetails.tsx

```diff
@@ -82,104 +118,181 @@ type ActionDetails = {
+    else
+        return {
+            type: ActionTypes.UNKNOWN,
+            name: 'invalid',
+            dialogTitle: 'invalid',
+            summary: 'invalid',
+            icon: <Error />,
+        };
+};
+
+const randomLockId = () => 'ui-' + Math.random().toString(36).substring(7);
+
+export const transformToBatch = (act: CartAction, m: string): BatchAction => {
+    if ('createEnvironmentLock' in act)
+        return {
+            action: {
+                $case: 'createEnvironmentLock',
+                createEnvironmentLock: {
+                    environment: act.createEnvironmentLock.environment,
+                    lockId: randomLockId(),
+                    message: m,
+                },
+            },
+        };
+    if ('createApplicationLock' in act)
+        return {
+            action: {
+                $case: 'createEnvironmentApplicationLock',
+                createEnvironmentApplicationLock: {
+                    environment: act.createApplicationLock.environment,
+                    application: act.createApplicationLock.application,
+                    lockId: randomLockId(),
+                    message: m,
+                },
+            },
+        };
+    if ('deleteEnvironmentLock' in act)
+        return {
+            action: {
+                $case: 'deleteEnvironmentLock',
+                deleteEnvironmentLock: {
+                    environment: act.deleteEnvironmentLock.environment,
+                    lockId: act.deleteEnvironmentLock.lockId,
+                },
+            },
+        };
+    if ('deleteApplicationLock' in act)
+        return {
+            action: {
+                $case: 'deleteEnvironmentApplicationLock',
+                deleteEnvironmentApplicationLock: {
+                    environment: act.deleteApplicationLock.environment,
+                    application: act.deleteApplicationLock.application,
+                    lockId: act.deleteApplicationLock.lockId,
+                },
+            },
+        };
+    if ('deploy' in act)
+        return {
+            action: {
+                $case: 'deploy',
+                deploy: {
+                    environment: act.deploy.environment,
+                    application: act.deploy.application,
+                    version: act.deploy.version,
+                    lockBehavior: LockBehavior.Ignore,
+                    ignoreAllLocks: false,
+                },
+            },
+        };
+    if ('prepareUndeploy' in act)
+        return {
+            action: {
+                $case: 'prepareUndeploy',
+                prepareUndeploy: {
+                    application: act.prepareUndeploy.application,
+                },
+            },
+        };
+    if ('undeploy' in act)
+        return {
+            action: {
+                $case: 'undeploy',
+                undeploy: {
+                    application: act.undeploy.application,
+                },
+            },
+        };
+    else return {};
```

Comment:
This is essentially an error, right? In that case I would:
```suggestion
    else return null;
```

You could then, for example, `.filter(isNonNullable)` to get rid of these invalid values. With:
```typescript
export const isNonNullable = <T>(value: T): value is NonNullable<T> => {
  return value !== undefined && value !== null;
};
```